src/app/repository/project-repository.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import {CpeApiService} from "./cpe-api.service";
+import {Project, RepositoryObject} from "./repository-model";
+
+
+interface iProjectRepo {
+  getProjectByID(id: string): any;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProjectRepositoryService implements iProjectRepo{
+
+  constructor(private apiService: CpeApiService) {
+  }
+
+  /**
+   * gets the list of projects via the api service. Returns it as a RepositoryObject array.
+   */
+  async getProjectsList():Promise<RepositoryObject[]> {
+    const pList: RepositoryObject[] = [];
+    const rJson: any[] = await this.apiService.getProjects().then(response => {
+      return response.json();
+    });
+    if(!Object.keys(rJson).length) { return []; } // if no projects exists
+    for (let i=0; i < rJson.length; i++){
+      const repObject = new RepositoryObject(rJson[i]['iri']);
+      repObject.label = rJson[i]['label'];
+      repObject.description = rJson[i]['description'];
+      pList.push(repObject);
+    }
+    return pList;
+  }
+
+  async isProjectExisting(projectId: string): Promise<boolean> {
+    return !!await this.getProjectByID(projectId)
+  }
+
+  /**
+   * gets a projects data via the api service. Returns a Project object.
+   */
+  async getProjectByID(id: string): Promise<Project | undefined> {
+    const rJson = await this.apiService.getProject(id).then(response => {
+      return response.json()
+    });
```

Comment:
You can shortcut it to: response => response.json()
json() returns a Promise with a JavaScript Object not an array

Reply:
wrong. json() returns a Promise<any>, in that very case it returns an any[].

Reply:
You are right about the .json() so you can keep the type definition.

Reply:
1. changed to httpclient. so no json() anymore. 
2. json() returns a Promise<any>, so also arrays like in the case of this fake api route